storage/ndb/src/kernel/vm/mt.cpp

```diff
@@ -4004,37 +4022,34 @@ thr_send_threads::handle_send_trp(TrpId trp_id,
   assert(m_trp_state[trp_id].m_thr_no_sender == NO_OWNER_THREAD);
   if (m_trp_state[trp_id].m_micros_delayed > 0)     // Trp send is delayed
   {
+    if (!is_send_thread(thr_no))
+    {
+      /**
+       * Assisting block threads need not care to send on a transporter
+       * which needs further delay. Handled by send threads.
+       */
+      return false;
+    }
     /**
      * The only transporter ready for send was a transporter that still
      * required waiting. We will only send if we have enough data to
      * send without delay.
      */
     if (m_trp_state[trp_id].m_send_overload)        // Pause overloaded trp
     {
+      /**
+       * Go to sleep and hopefully the overload has cleared when we wake up
+       * again.
+       */
       return false;
     }
 
-    if (mt_get_send_buffer_bytes(trp_id) >= MAX_SEND_BUFFER_SIZE_TO_DELAY)
```

Comment:
Yep, MaxSendDelay have been made ineffective, it is now always on, so not configurable anymore

Reply:
If it is now always on, then I think we should ensure that the trp must wait for at least sendDelay before the actual sending, remove it would cause no wait anymore even if sendDelay is always on.

Reply:
Yep, working on a redesign of this part. Will discuss it more in the office tomorrow.